packages/flutter/test/material/text_selection_test.dart

```diff
@@ -581,6 +581,39 @@ void main() {
         matchesGoldenFile('transparent_handle.png'),
       );
     });
+
+    testWidgets('works with 3 positional parameters', (WidgetTester tester) async {
+      await tester.pumpWidget(Theme(
+        data: ThemeData(
+          textSelectionTheme: const TextSelectionThemeData(
+            selectionHandleColor: Color(0x550000AA),
+          ),
+        ),
+        child: Builder(
+          builder: (BuildContext context) {
+            return Container(
+              color: Colors.white,
+              height: 800,
+              width: 800,
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 250),
+                child: FittedBox(
+                  child: materialTextSelectionControls.buildHandle(
+                    context, TextSelectionHandleType.right, 10.0,
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      ));
+
+      expect(
+        // CustomPaint is inserted by the Material text selection controls.
+        find.byType(CustomPaint),
+        findsOneWidget,
+      );
```

Comment:
@Hixie do you think this is a valid test case?

I thought about just using `test`, but then I could not pass an actual `BuildContext`.

Reply:
Does it work if you create a `MaterialTextSelectionControl` subclass and override its `buildHandle`?

Reply:
@LongCatIsLooong Do you mean if the test breaks in that case? Otherwise, I mean of course, the method syntax must stay the same. So as long as you return a CustomPaint..

Reply:
No I mean if you want to use `test` instead of `testWidgets`, would it be possible to just test with a custom `MaterialTextSelectionControl` subclass with an empty `buildHandle` implementation? To acquire a `BuildContext` you can either use something like this:https://github.com/flutter/flutter/blob/03034e9e4aabe4ae99cc21a7010f623c69842fd3/packages/flutter/test/widgets/slivers_test.dart#L1130-L1133 or instantiate a random `Element`.

Reply:
@LongCatIsLooong ok, I knew this is possible, but I asked myself: is this preferrable? So my thought was: just use `testWidgets` because it is _easier_ to make an actual call. Of course I can mock it, but that would be more complicated than just using `WidgetTester` 🤔

Reply:
In that case we can just go with this. Nit: could you replace the `expect` at the end with comments that says this should compile, to make sure people don't simply change the tests (like I did)? The expect there is actually slightly misleading.

Reply:
@LongCatIsLooong I agree - I changed it 👍